typing/path.ml

```diff
@@ -88,6 +94,52 @@ let rec scope = function
   | Pdot(p, _) | Pextra_ty (p, _) -> scope p
   | Papply(p1, p2) -> Int.max (scope p1) (scope p2)
 
+let rec contains id = function
+    Pident id' -> Ident.same id' id
+  | Pdot(p, _) | Pextra_ty(p, _) -> contains id p
+  | Papply(p1, p2) -> contains id p1 || contains id p2
+
+let subst id_map p =
+  let exception Unchanged in
```

Comment:
For future reference, this can also be done with

```ocaml
let changed = ref false in
...
   List.find (fun (i, _) -> Ident.same i id && (changed := true; true)) id_map
...
let p' = aux p in
if !changed then p' else p
```